sesman.el

```diff
@@ -319,6 +332,30 @@ If SORT is non-nil, sort in relevance order."
 (defun sesman--lnk-value (lnk)
   (nth 2 lnk))
 
+(defun sesman--follow-symlink-p (filename)
+  "FILENAME predicate that tries to predict whether `find-file' will
+follow symlinks (t) or not (nil)."
+  (or find-file-visit-truename
+      (and vc-follow-symlinks
+           (let ((truename (sesman--file-truename filename)))
+             (and truename
+                  (not (equal truename filename))
+                  (vc-backend truename))))))
+
+(defvar sesman--file-truename-cache (make-hash-table :test #'equal))
+(defun sesman--file-truename (filepath)
+  "Optimizing `file-truename' wrapper."
+  (let ((exp-fp (expand-file-name filepath)))
+   (cond
+    ((string= exp-fp default-directory)
```

Comment:
Can `default-directory` be not expanded?

Reply:
Yes. Often actually.

Reply:
@vspinu thx.